### ISSUE TITLE
Setting up demo notebooks with JupyterLite

### ISSUE THREAD

**Issue by BSchilperoort** [AI4S2S/s2spy#63]
> JupyterLite is a JupyterLab distribution that runs entirely in the browser built from the ground-up using JupyterLab components and extensions.
https://jupyterlite.readthedocs.io/en/latest/

JupyterLite would allow us to make interactive demo-notebooks that potential users can run immediately _on their own machines_. It makes use of Pyodide, which is a port of CPython to WebAssembly, meaning that it runs inside the browser without any installation. 

I tried out setting up a jupyter lite notebook with s2spy, and it works without any problems;
![image](https://user-images.githubusercontent.com/12114825/182362974-12132f78-13ea-4b1d-abe0-9987f9db4061.png)

_(it will even work on people's phones)_

**Comment by Peter9192** [comment]
Cool! That's already way more advanced than last time I checked. Does it also work when you're using pandas/xarray/sklearn dependencies?

**Comment by Peter9192** [comment]
Ideally, I'd like to have such interactive examples in the documentation instead of in a separate set of notebooks.

**Comment by BSchilperoort** [comment]
> Cool! That's already way more advanced than last time I checked. Does it also work when you're using pandas/xarray/sklearn dependencies?

Yup. It all works. Well, except xarray's optional netcdf4 dependency: we cannot read netcdf files at the moment.



> Ideally, I'd like to have such interactive examples in the documentation instead of in a separate set of notebooks.

Well, we can put those in using, e.g., [pyscript ](https://pyscript.net/), see [Peter Wang's Pycon keynote](https://www.youtube.com/watch?v=qKfkCY7cmBQ). I am not sure about the current status of integrating jupyterlite and readthedocs.

**Comment by BSchilperoort** [comment]
> Ideally, I'd like to have such interactive examples in the documentation instead of in a separate set of notebooks.

Turns out, we can: https://github.com/jupyterlite/jupyterlite-sphinx
An example of this is on their docs page: https://jupyterlite-sphinx.readthedocs.io/en/latest/

**Comment by Peter9192** [comment]
That is absolutely awesome! Though a bit slow perhaps.
